algorithms/src/std_algorithms/Kokkos_RandomAccessIterator.hpp

```diff
@@ -0,0 +1,213 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_RANDOM_ACCESS_ITERATOR_HPP
+#define KOKKOS_RANDOM_ACCESS_ITERATOR_HPP
+
+#include <iterator>
+#include "Kokkos_Macros.hpp"
+#include <Kokkos_View.hpp>
+#include "./Kokkos_Constraints.hpp"
+
+namespace Kokkos {
+namespace Experimental {
+namespace Impl {
+
+template <class T>
+class RandomAccessIterator;
+
+template <class DataType, class... Args>
+class RandomAccessIterator< ::Kokkos::View<DataType, Args...> >
+    : public std::iterator<
+          std::random_access_iterator_tag,
+          typename ::Kokkos::View<DataType, Args...>::value_type, ptrdiff_t,
+          typename ::Kokkos::View<DataType, Args...>::pointer_type,
+          typename ::Kokkos::View<DataType, Args...>::reference_type> {
+ public:
+  using view_type       = ::Kokkos::View<DataType, Args...>;
+  using iterator_type   = RandomAccessIterator<view_type>;
+  using difference_type = ptrdiff_t;
+  using value_type      = typename view_type::value_type;
+  using reference       = typename view_type::reference_type;
+
+  static_assert(view_type::rank == 1 &&
+                    (std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutLeft>::value ||
+                     std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutRight>::value ||
+                     std::is_same<typename view_type::traits::array_layout,
+                                  Kokkos::LayoutStride>::value),
+                "RandomAccessIterator only supports 1D Views with LayoutLeft, "
+                "LayoutRight, LayoutStride.");
+
+  RandomAccessIterator() = default;
+
+  explicit KOKKOS_INLINE_FUNCTION RandomAccessIterator(const view_type view)
+      : m_view(view) {}
+  explicit KOKKOS_INLINE_FUNCTION RandomAccessIterator(const view_type view,
+                                                       ptrdiff_t current_index)
+      : m_view(view), m_current_index(current_index) {}
+
+  KOKKOS_INLINE_FUNCTION
+  iterator_type& operator=(const iterator_type& it) {
+    m_view          = it.m_view;
+    m_current_index = it.m_current_index;
+    return *this;
+  }
```

Comment:
Why did you define it at all?

Reply:
Removed (note: using `default` works fine as well).